### PR TITLE
Add Service Account JSON Auth

### PR DESCRIPTION
**Pull request by michelvocks**
I tested it and it seems to work. I noticed a few errors during the fetch that looks like this:

```
{
  "level": "error",
  "error": "googleapi: Error 403: Cloud Functions API has not been used in project 253264496340 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/cloudfunctions.googleapis.com/overview?project=253264496340 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry., accessNotConfigured",
  "provider": "gcp",
  "time": "2021-07-14T17:49:57+02:00",
  "message": "received error fetching"
}
{
  "level": "error",
  "error": "googleapi: got HTTP response code 404 with body: <!DOCTYPE html>\n<html lang=en>\n  <meta charset=utf-8>\n  <meta name=viewport content=\"initial-scale=1, minimum-scale=1, width=device-width\">\n  <title>Error 404 (Not Found)!!1</title>\n  <style>\n    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}\n  </style>\n  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>\n  <p><b>404.</b> <ins>That’s an error.</ins>\n  <p>The requested URL <code>/v1beta1/projects/cq-playground/location/-/registrations?alt=json&amp;pageToken=&amp;prettyPrint=false</code> was not found on this server.  <ins>That’s all we know.</ins>\n",
  "provider": "gcp",
  "time": "2021-07-14T17:49:58+02:00",
  "message": "received error fetching"
}
```

But I think these are coming from disabled APIs?

### CODE REVIEW

File: client/config.go [hunk 1]
```diff
@@ -2,8 +2,9 @@ package client
 
 // Config defines Provider Configuration
 type Config struct {
-	ProjectFilter string   `hcl:"project_filter,optional"`
-	ProjectIDs    []string `hcl:"project_ids,optional"`
+	ProjectFilter         string   `hcl:"project_filter,optional"`
+	ProjectIDs            []string `hcl:"project_ids,optional"`
+	ServiceAccountKeyJSON string   `hcl:"service_account_key_json,optional"`
```
Comment (roneli) [inline client/config.go:7]: we can make this a block and then marshal that block into a json and pass it to the client, but I am not sure if that's prettier than the actual json itself. wdyt?

**Review by roneli — APPROVED** [review]
LGTM, had one comment tell me what you think.

**Comment by roneli** [comment]
@fdistorted do you know of the errors above?

**Comment by fdistorted** [comment]
I have also faced the `Functions API is not enabled` error. I also faced similar errors related to other services. I have just enabled them and it solved the error. I didn't think about it as an issue.

File: client/config.go (hunk 1 above)
Comment (michelvocks) [reply client/config.go:7]: How would that look like in the `config.hcl` file? I guess you would have to translate the secrets file (which is json) to HCL?

**Pull request merged**